src/git-sync

```diff
@@ -31,6 +31,13 @@ function preconditions {
 }
 
 
+function set_previous_branch {
+  if [ "$IN_SUB_FOLDER" = false ]; then
+    set_initial_previous_branch_as_previous_branch
```

Comment:
Could this crazy long method (which isn't defined in this PR) be replaced with `set_previous_branch $INITIAL_PREVIOUS_BRANCH_NAME` or something comparable?